eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -1983,6 +1985,28 @@ default void appendString(Appendable appendable, String separator)
     <V> Multimap<V, T> groupBy(Function<? super T, ? extends V> function);
 
     /**
+     * This method will count the number of occurrences of each value calculated by applying the
+     * function to each element of the collection.
+     *
+     * @since 8.2
+     */
+    default <V> Bag<V> countBy(Function<? super T, ? extends V> function)
+    {
+        return this.asLazy().<V>collect(function).toBag();
```

Comment:
I see what you mean. If someone has implemented RichIterable, and it doesn't extend anything like AbstractMutableCollection, or implement MutableCollection or ImmutableCollection, then it fails to compile. But if we do add a method body, then we'd want to delete the body in the next major version I'd think. And yet the method is not deprecated. Just its body is probably dead code.